docs/vhlo.md

```diff
@@ -0,0 +1,128 @@
+# The VHLO Dialect
+
+## What is the VHLO Dialect?
+
+The VHLO (Versioned StableHLO) Dialect is used for serialization and stability.
+It provides a snapshot of the StableHLO dialect at a given point in time,
+denoted by version number.
+
+VHLO is an **add-only dialect** with **versioned ops**, which means that once an
+op is added to the dialect, it cannot be modified in any way that impact the
+ops semantics.
+
+Any changes to an op require a new version of an op to be added to the dialect.
+For example, if a hypothetical `my_op` was added to StableHLO in 0.9.0, but
+was changed in 0.11.0, we would have the following in VHLO:
+
+```tablegen
+// This represents the StableHLO version of the op from 0.9.0 -> 0.10.0
+def VHLO_MyOpV1 : VHLO_Op<"my_op", "0.9.0", "0.10.0"> {
+  let arguments = (ins
+    VHLO_AnyType:$operand
+  );
+  let results = (outs VHLO_AnyType:$result);
+}
+
+// This represents the StableHLO version of the op from 0.11.0 -> current
+def VHLO_MyOpV2 : VHLO_Op<"my_op_v2", "0.11.0", "current"> {
+  let arguments = (ins
+    VHLO_AnyType:$operand,
+    VHLO_AnyAttr:$attr  // New attribute added to StableHLO in 0.11.0
+  );
+  let results = (outs VHLO_AnyType:$result);
+}
+```
+
+Note that under this schema, the latest version of an op directly represents
+the StableHLO equivalent of the op.
+
+## Why is this necessary?
```

Comment:
How about "Why is VHLO useful?". Strictly speaking, VHLO is not necessary, because there are other successful systems that don't have anything like this.